src/controllers/update/overseas.entity.change.controller.ts

```diff
@@ -49,3 +52,33 @@ export const post = async (req: Request, resp: Response, next: NextFunction) =>
     next(errors);
   }
 };
+
+export const resetChangeData = async (req: Request, appData: ApplicationData) => {
+  if (appData){
+    appData.who_is_registering = undefined;
+    appData.overseas_entity_due_diligence = undefined;
+    appData.beneficial_owners_individual = undefined;
+    appData.beneficial_owners_corporate = undefined;
+    appData.beneficial_owners_government_or_public_authority = undefined;
+    appData.managing_officers_corporate = undefined;
+    appData.managing_officers_individual = undefined;
+    appData.beneficial_owners_statement = undefined;
+    appData.payment = undefined;
```

Comment:
The OE entity could be changed also so do:
const companyProfile = await getCompanyProfile(req, appData.entityNumber);
await addOeToApplicationData(req, appData, appData.entityNumber, companyProfile);

Reply:
appData.due_diligence = undefined
appData.overseas_entity_due_diligence = undefined

// appData.presenter is entered but is ok
// maybe so we don't forget in future
appData.trusts = undefined

Reply:
appData.update.bo_mo_data_fetched = false;

Reply:
But won't the OE change be handled before no_change journey. The reset will take place on the OE query page I suppose? I would think only the data from no change page is required to reset?

Reply:
// appData.update.date_of_creation, appData.entity_name and appData.entity_number won't be affected.
// appData.update.no_change will be toggled anyway
// appData.filing_date wil not be affected either

appData.update.registrable_benefitial_owner needs to be reset to undefined

Reply:
I did not mean OE change as in switching to another OE altogether. I meant the user changing something in the OE data - like the email address or other change links on the "OE revirew" page. So re-load the same OE.